x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptFieldTypeTestCase.java

```diff
@@ -67,6 +67,14 @@ protected static QueryShardContext mockContext(boolean allowExpensiveQueries) {
         return mockContext(allowExpensiveQueries, null);
     }
 
+    protected boolean supportsTermQueries() {
```

Comment:
Maybe it'd be better to make a new superclass that doesn't have these methods?

Reply:
The problem is that we already have `AbstractNonTextScriptFieldTypeTestCase` that extends this class. Those methods are worthy running so adding a superclass you will miss that. My feeling is that we need an abstract class for every type of query and field types will implement all the ones they support?

Reply:
So instead of an abstract superclass for the type of field it is per query? I could see that being better, maybe. I'm ok with this as is if we plan to refactor it soon enough anyway.